_posts/2024-08-23-GSoC-CSL.md

```diff
@@ -0,0 +1,65 @@
+---
+title: JabRef's new CSL Integration for LibreOffice
+tags: [gsoc, csl]
+author: subhramit
+---
+
+Hi, I am Subhramit. I am a Google Summer of Code (GSoC) student at JabRef this year, and I'll be walking you through JabRef's new CSL Integration for LibreOffice.
+
+## Background
+
+As a student, I first came across pre-defined citation formats during my freshman year of college, when our professor instructed that the references in our project report must follow the "APA style".  
+
+I remember manually generating and copying citations in that style from some citation generation website, and oh was it tedious. Whether you are a student like me or a seasoned researcher in academia, you must have grappled with various citation styles throughout your academic journey.
+
+[Citation Style Language](https://citationstyles.org/) (CSL) is a popular open-source specification language that standardizes the formatting of citations and bibliographies. It supports thousands of standard citation styles used in academia, including *American Psychological Association* (APA), *Modern Language Association* (MLA), *Chicago Manual of Style* (CMS), *Institute of Electrical and Electronics Engineers* (IEEE), *Springer - Lecture Notes in Computer Science* and *Vancouver* to name a few. If you've ever cited a source in an academic paper, chances are you've used a CSL-based style.
+
+For several years, JabRef users have been requesting support for CSL in its LibreOffice integration. Tackling this problem statement this summer as a part of my project, I am excited to announce that this highly anticipated feature is now available!
+
+## What's New?
+
+With this integration, JabRef users can now:
+
+1. **Choose a CSL Style**: Select from a vast array of CSL styles to match the requirements of your academic work.
+2. **Cite with Flexibility**: Insert citations as well as in-text citations (similar to LaTeX's `\cite` and `\citet` commands) directly into your LibreOffice document.
```

Comment:
"Insert usual citiations"...

It reads strange that no attribute for "citations" first, but we have no word for "non-intext.citations", do we?

Replace "LaTeX's" by "natbib's" and link https://ctan.org/pkg/natbib.

Reply:
"Usual citations" sounds odd.
How about this ^ (check changes).

Reply:
If not this, the other alternative - "insert citations directly into..." (remove in-text, as citations is generic and can encompass all types).

Reply:
Then also remove link to natbib, because there is no distinction any more

Reply:
Sorry, what I meant was, I have already changed it to:
"**Cite with Flexibility**: Insert citations (similar to natbib's `\cite`) as well as in-text citations (similar to natbib's and `\citet`) directly into your LibreOffice document."

If we don't like this, then we can go with:
"**Cite with Flexibility**: Insert citations directly into your LibreOffice document."
(with natbib removed).

Reply:
I realized I forgot to add the link. Will wait for your approval (if we choose to go with the first one).

Reply:
I think Thilo isn't tagged properly so @ThiloteE

Reply:
@koppor Side comment: I do think case 2 looks cleaner (and more digestible for a general audience, even for students like me who didn't use LaTeX during the college course, the story of which I mentioned in the blog's introductory paragraph).

Reply:
I suggest trying to avoid filler words and non-words, such as "flexible", especially for documentation. I think the sentence is good enough for the Blog, though. While I have used LaTeX and bibaltex, I have never used the Natbib package before. I think JabRef and CSL define the way they cite themselves independently from Natbib, and, yes, while it can help users that are used to that package help understand the new features it might be easier to understand, if the feature is explained, with verbs instead of nouns.

Reply:
Which sentence, Thilo?
Your suggestion would be to remove mentioning natbib (or LaTeX) altogether?
Are you referring to case 1, case 1 with natbib removed, or case 2?

Reply:
I have updated it to:
"**Cite with ease**: Insert citations directly into your LibreOffice document."

Reply:
I am not sure what case 1 or case 2 are.

Ideally, I would prefer splitting up Cite and Cite-in-Text into separate sections, but it's fine, if that is explained more in detail in the documentation. 
The way it is now with natbib is ok. We have many LaTeX users, so Natbib might ring a bell.

Reply:
Can you check if the current sentence is okay? Or if I should revert it and bring back the distinctions and natbib refrences?

Reply:
To be honest, I don't have a strong opinion here. Your choice :-)

Reply:
Alright.
@koppor Then I suggest the current going with the current:
"Cite with ease: Insert citations directly into your LibreOffice document."